pandas/core/indexes/base.py

```diff
@@ -4762,8 +4779,9 @@ def _get_engine_target(self) -> ArrayLike:
             # GH#45652 much more performant than ExtensionEngine
             return vals._ndarray
         if type(self) is Index and isinstance(self._values, ExtensionArray):
-            # TODO(ExtensionIndex): remove special-case, just use self._values
-            return self._values.astype(object)
+            if not isinstance(self._values, BaseMaskedArray):
```

Comment:
an this be combined with the previous condition?

Reply:
Done, found this a bit easier to read, but no strong opinion